### DISCUSSION TITLE
Problems uploading nupkg

### DISCUSSION THREAD

**Discussion by box-turtle** [microsoft/Microsoft-Fabric-workload-development-sample#22] [Help] [answered]
We are getting a validation error trying to upload our workload. It passes the xsd validation, but fails with this error when we try to upload.

```
{
    "error": {
        "code": "ExtensionsInvalidPackage",
        "pbi.error": {
            "code": "ExtensionsInvalidPackage",
            "details": [
                {
                    "code": "Reason",
                    "detail": {
                        "type": 1,
                        "value": "InvalidParameter"
                    }
                },
                {
                    "code": "Target",
                    "detail": {
                        "type": 1,
                        "value": "WorkloadManifest.xml"
                    }
                }
            ],
            "exceptionCulprit": 1
        }
    }
}
```

So we know the problem is with WorkloadManifest.xml but I can't figure out what the issue is. 

Our verified domain is `nonprod.aiunlimited.teracloud.ninja`

Here is our WorkloadManifest.xml

```
<?xml version="1.0" encoding="utf-8" ?>
<WorkloadManifestConfiguration xmlns:xsi= "http://www.w3.org/2001/XMLSchema-instance" SchemaVersion="1.0.0">
  <Workload WorkloadName="Org.Teradata" HostingType="Remote">
    <Version>1.0.0</Version>
    <RemoteServiceConfiguration>
      <CloudServiceConfiguration>
        <Cloud>Public</Cloud>
        <AADApp>
          <AppId>45f3da91-fda8-42ff-9f1e-b06ce9b2c2cd</AppId> <!-- Your application ID -->
          <RedirectUri>https://feserver.nonprod.aiunlimited.teracloud.ninja/close</RedirectUri>
          <ResourceId>https://nonprod.aiunlimited.teracloud.ninja/feserver/session-manager/Org.Teradata/preview</ResourceId>
        </AADApp>
        <Endpoints>
          <ServiceEndpoint>
            <Name>Workload</Name>
            <Url>https://session-manager.nonprod.aiunlimited.teracloud.ninja/v1</Url>
            <IsEndpointResolutionService>false</IsEndpointResolutionService>
          </ServiceEndpoint>
          <ServiceEndpoint>
            <Name>Frontend</Name>
            <Url>https://feserver.nonprod.aiunlimited.teracloud.ninja</Url>
            <IsEndpointResolutionService>false</IsEndpointResolutionService>
          </ServiceEndpoint>
        </Endpoints>
      </CloudServiceConfiguration>
    </RemoteServiceConfiguration>
  </Workload>
</WorkloadManifestConfiguration>

```

**Comment by anitamayorov** [comment]
Hi @box-turtle,

Can you please provide a requestId for the failed Upload network call?

**Comment by box-turtle** [comment]
requestId:

```
c1fe8128-9dc5-72a7-1205-675219d8216e
```

**Reply by anitamayorov** [reply]
Hi @box-turtle,

It looks like your WorkloadManifest.xml file location is not as expected, can you please verify your NuGet package looks like this: 
![image](https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/assets/80410523/9908ecfe-4bbf-47f8-aef6-2a70c7ac834e)

**Reply by box-turtle** [reply]
It seems to be in the correct location in our NuGet package?

<img width="697" alt="image" src="https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/assets/3999743/fdca8f43-ae1c-437d-8708-db2af3afab54">

Had to rename the nupkg to zip to upload it, but here it is:

[ManifestPackage.1.0.0.zip](https://github.com/user-attachments/files/15991206/ManifestPackage.1.0.0.zip)

**Reply by box-turtle** [reply]
really appreciate your help on this!

**Reply by box-turtle** [reply]
is it possible that the fact our verified domain `nonprod.aiunlimited.teracloud.ninja` has 4 segments instead of 3 like the examples could be causing a problem?

**Reply by box-turtle** [reply]
Just to give more context - our back end is not a .net app deployed according to the [instructions here](https://learn.microsoft.com/en-us/fabric/workload-development-kit/azure-web-app-deployment-tutorial).

But our BE does live at https://session-manager.nonprod.aiunlimited.teracloud.ninja/v1, so uses a subdomain of our verified domain.

We did deploy our FE following those instructions and mapped the verified domain.

**Reply by anitamayorov** [reply, marked as answer]
Hi @box-turtle ,

I looked at the package you sent - 
1. You have there a folder inside the zip, and only inside that folder are the expected files - is this your original zip you've been trying to upload?
![image](https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/assets/80410523/10dc1683-c581-4546-9d07-d6eae9b757b8)

2. Inside the folder, you have FrontendManifest.json file - this is not needed since all of your workload's frontend resources are located under 'FE' folder

When I zipped the inner folder's content - I was able to upload the package successfully. 
I am not aware of a limitation on segments for the verified domain, I will verify on our side that it's indeed OK. Either way, if such, the error you will see in network traces will be different, your current error indicates missing files or wrong location in package. 

Can you try package your files again without any subfolders, so that the zip will contain two subfolders - BE and FE ? 
![image](https://github.com/microsoft/Microsoft-Fabric-workload-development-sample/assets/80410523/c3444c59-2665-404e-bc34-2d6e828f549d)

**Reply by box-turtle** [reply]
thanks so much! Worked for me this morning!
